### PR TITLE
feat: Enums in JVM

### PR DESCRIPTION
**Pull request by tomdaffurn**
For #2514 

Tasks: 
- [x] Test cases for all enum variants
- [x] Generate Java classes for enums in schema
- [x] Serialize/deserialise enums for JVM verb server
- [x] Serialize/deserialise enums for JVM verb clients
- [x] Generate schema for enums in Java code
- ~~Generate schema a nd classes for enums in Kotlin code~~ #3062
- ~~Update comment extraction code to include enums~~ #3061

### CODE REVIEW

File: jvm-runtime/ftl-runtime/common/deployment/src/main/java/xyz/block/ftl/deployment/JVMCodeGenerator.java [hunk 1]
```diff
@@ -26,6 +29,7 @@ public abstract class JVMCodeGenerator implements CodeGenProvider {
 
     public static final String PACKAGE_PREFIX = "ftl.";
     public static final String TYPE_MAPPER = "TypeAliasMapper";
+    private static final Logger log = LoggerFactory.getLogger(JVMCodeGenerator.class);
```
Comment (stuartwdouglas) [inline jvm-runtime/ftl-runtime/common/deployment/src/main/java/xyz/block/ftl/deployment/JVMCodeGenerator.java:32]: slf4j

File: jvm-runtime/ftl-runtime/common/deployment/src/main/java/xyz/block/ftl/deployment/ModuleProcessor.java [hunk 2]
```diff
@@ -44,11 +44,10 @@
 import xyz.block.ftl.runtime.VerbRegistry;
 import xyz.block.ftl.runtime.config.FTLConfigSourceFactoryBuilder;
 import xyz.block.ftl.runtime.http.FTLHttpHandler;
-import xyz.block.ftl.v1.schema.Ref;
 
 public class ModuleProcessor {
 
-    private static final Logger log = Logger.getLogger(ModuleProcessor.class);
+    private static final Logger log = LoggerFactory.getLogger(ModuleProcessor.class);
```
Comment (stuartwdouglas) [inline jvm-runtime/ftl-runtime/common/deployment/src/main/java/xyz/block/ftl/deployment/ModuleProcessor.java:50]: And here

File: jvm-runtime/ftl-runtime/common/runtime/src/test/java/xyz/block/ftl/runtime/Animal.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package xyz.block.ftl.runtime;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import xyz.block.ftl.Enum;
+
+@Enum
+public interface Animal {
+    @JsonIgnore
```
Comment (stuartwdouglas) [inline jvm-runtime/ftl-runtime/common/runtime/src/test/java/xyz/block/ftl/runtime/Animal.java:9]: I wonder if we can eventually remove the need for these annotations.

**Review by stuartwdouglas — APPROVED** [review]
LGTM, except for the logging changes.

File: jvm-runtime/ftl-runtime/common/deployment/src/main/java/xyz/block/ftl/deployment/DatasourceProcessor.java [hunk 4]
```diff
@@ -4,6 +4,9 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import org.slf4j.Logger;
```
[the remark this answers is not in the document]
Comment (tomdaffurn) [reply jvm-runtime/ftl-runtime/common/deployment/src/main/java/xyz/block/ftl/deployment/DatasourceProcessor.java]: Cool, made that change

File: jvm-runtime/ftl-runtime/common/runtime/src/test/java/xyz/block/ftl/runtime/Animal.java (hunk 3 above)
Comment (tomdaffurn) [reply jvm-runtime/ftl-runtime/common/runtime/src/test/java/xyz/block/ftl/runtime/Animal.java:9]: Yeah, maybe we can customise the serialisation further to ignore them

**Pull request merged**